task_geo/data_sources/covid/fr_covidata/fr_covidata.py

```diff
@@ -0,0 +1,257 @@
+    https://github.com/opencovid19-fr/data
+"""
+
+import io
+
+import pandas as pd
+import requests
+
+url = (
+    'https://raw.githubusercontent.com/opencovid19-fr/'
+    'data/master/dist/chiffres-cles.csv'
+)
+
+
+def fr_covidata():
+    """Data Source for the French COVID-19 Data.
+    Arguments:
+        None
+    Returns:
+        pandas.DataFrame
+    """
+    df = fr_covidata_connector()
+    return fr_covidata_formatter(df)
+
+
+def fr_covidata_connector():
+    """Extract data from OpenCOVID19-fr's Github repository.
+    Description:
+        - Downloads the URL's data in a Unicode CSV Format
+        - Unicode CSV Format: ACS 5Y UTF-8
+    Returns:
+        dataset (DataFrame with CSV Data)
+    """
+
+    urlData = requests.get(url).content
+
+    dataset = pd.read_csv(io.StringIO(urlData.decode('utf-8')))
+    return dataset
+
+
+def fr_covidata_formatter(dataset):
+    """Formatter for FR COVID-19 Data.
+    Arguments:
+        dataset(pandas.DataFrame): Data as returned by fr_covidata_connector.
+    Description:
+        - Drop unnecessary rows with irrelevant regions' info and only keep
+        info related to subregions in Metropolitan France, as well as
+        repetitive data
+        - Check the dataset for instances where there are more than one source
+        of data in the same subregion for the same date, then complement all
+        the sources information, and take the highest value in case there are
+        different values for the same column, while aggregating the sources
+        info
+        - Rename/Translate the column titles, and add a country column (France)
+    Returns:
+        frcovidata(pandas.DataFrame)
+    """
+
+    dataset = dataset[dataset.granularite != 'region']
+    dataset = dataset[dataset.granularite != 'monde']
+    dataset = dataset[dataset.granularite != 'pays']
+    dataset = dataset[dataset.granularite != 'collectivite-outremer']
+    dataset = dataset[dataset.maille_code != 'DEP-971']
+    dataset = dataset[dataset.maille_code != 'DEP-972']
+    dataset = dataset[dataset.maille_code != 'DEP-973']
+    dataset = dataset[dataset.maille_code != 'DEP-974']
+    dataset = dataset[dataset.maille_code != 'DEP-976']
+    dataset = dataset.drop(['depistes'], axis=1)
+    dataset = dataset.drop(['granularite'], axis=1)
+    dataset = dataset.drop_duplicates(
+        subset=['date', 'maille_code', 'cas_confirmes', 'deces',
+                'reanimation',
+                'hospitalises', 'gueris'], keep=False)
+
+    # Reset indices:
+    dataset = dataset.reset_index()
+    dataset = dataset.drop(['index'], axis=1)
+
+    # Turn source columns' values type to string:
+    dataset['source_nom'] = dataset['source_nom'].astype(str)
+    dataset['source_url'] = dataset['source_url'].astype(str)
+    dataset['source_archive'] = dataset['source_archive'].astype(str)
+    dataset['source_type'] = dataset['source_type'].astype(str)
```

Comment:
This can be done in a more elegant way with just to lines:

```python
str_columns = [...]
dataset[str_columns] = dataset[str_columns].astype(str)
```

But also, which type they were before you casted them to string? Please note that for pandas all non-numeric and non-datetime columns are shown as object, so check if this is really necessary at all.